pkg/windows/calico.go

```diff
@@ -52,147 +53,226 @@ const (
+	return nil
+}
 
-	hc, err := helm.NewFactoryFromConfig(restConfig)
+func (c *Calico) GetConfig() *CNIConfig {
+	return c.CalicoCNICfg
+}
+
+
+// initializeConfig sets the default configuration in CNIConfig
+func (c *Calico) initializeConfig(ctx context.Context, nodeConfig *daemonconfig.Node, restConfig *rest.Config) error {
+	c.CalicoCNICfg = &CNIConfig{NetworkName: "Calico", BindAddress: nodeConfig.AgentConfig.NodeIP}
+
+	platformType, err := getPlatformType()
 	if err != nil {
-		return nil, err
+		return err
 	}
 
-	if err := getDefaultConfig(cfg, dataDir, nodeConfig); err != nil {
-		return nil, err
+	c.CalicoCNICfg.CalicoConfig = &CalicoConfig{
+		Name:                  "Calico",
+		Hostname:              nodeConfig.AgentConfig.NodeName,
+		NodeNameFile:          filepath.Join("c:\\", c.DataDir, "agent", CalicoNodeNameFileName),
+		KubeNetwork:           "Calico.*",
+		Mode:                  "vxlan",
+		ServiceCIDR:           nodeConfig.AgentConfig.ServiceCIDR.String(),
+		DNSServers:            nodeConfig.AgentConfig.ClusterDNS.String(),
+		DNSSearch:             "svc." + nodeConfig.AgentConfig.ClusterDomain,
+		DatastoreType:         "kubernetes",
+		NetworkingBackend:     "vxlan",
+		Platform:              platformType,
+		StartUpValidIPTimeout: 90,
+		LogDir:                "",
+		IP:                    "autodetect",
+		IPAutoDetectionMethod: "first-found",
+		Felix: FelixConfig{
+			Metadataaddr:    "none",
+			Vxlanvni:        "4096",
+			MacPrefix:       "0E-2A",
+			LogSeverityFile: "none",
+			LogSeveritySys:  "none",
+		},
+		CNI: CalicoCNIConfig{
+			BinDir:   nodeConfig.AgentConfig.CNIBinDir,
+			ConfDir:  nodeConfig.AgentConfig.CNIConfDir,
+			IpamType: "calico-ipam",
+			Version:  "0.3.1",
+		},
 	}
 
-	cfg.CalicoConfig.KubeConfig = calicoKubeConfig
-	if err := getCNIConfigOverrides(cfg, hc); err != nil {
-		return nil, err
+	c.CalicoCNICfg.CalicoConfig.KubeConfig, err = c.createCalicoKubeConfig(ctx, restConfig)
+	if err != nil {
+		return err
 	}
 
-	agent.NetworkName = CalicoHnsNetworkName
+	return nil
+}
 
-	if err := createNodeNameFile(cfg, dataDir); err != nil {
-		return nil, err
+// writeConfigFiles writes the three required files by Calico
+func (c *Calico) writeConfigFiles(CNIConfDir string, NodeName string) error {
+	// parse the two config files required by Calico
+	templates, err := parseCalicoTemplatesFromConfig(c.CalicoCNICfg)
+	if err != nil {
+		return err
+	}
+	
+	// create calico kubeconfig, calico config and node name files
+	if err = util.WriteFile(filepath.Join("c:\\", c.DataDir, "agent", CalicoKubeConfigName), templates[calicoKubeConfigTemplate]); err != nil {
+		return err
+	}
+	if err = util.WriteFile(filepath.Join(CNIConfDir, CalicoConfigName), templates[calicoConfigTemplate]); err != nil {
+		return err
+	}
+    if err = util.WriteFile(c.CalicoCNICfg.CalicoConfig.KubeConfig.Path, NodeName); err != nil {
```

Comment:
the indentation here appears to be off? I'm also confused what's going on, we're writing a file containing the NodeName where the calico kubeconfig should be?

Reply:
For some reason, VisualStudio does not digest well files starting with the lines:
```
//go:build windows
// +build windows
```
so the automatic fmt I always get is not working anymore. I `gofmt` it manually and it should be fixed now.

It seems it is reusing the path of Kubeconfig.... but why it does not use that when creating the kubeconfig? I need to change this and maybe use a more general path name

Reply:
yeah, I have the same problem with vim - unless I override the GOARCH when starting vim it won't autoformat or autocomplete in the windows-tagged files.